connectivity/drivers/emac/TARGET_STM/stm32xx_emac.cpp

```diff
@@ -242,20 +242,14 @@ static void MPU_Config(void)
 
 #endif
 
-
-
 /**
- * Ethernet Rx Transfer completed callback
+ * IRQ Handler
  *
  * @param  heth: ETH handle
  * @retval None
  */
-void HAL_ETH_RxCpltCallback(ETH_HandleTypeDef *heth)
+MBED_WEAK void STM_HAL_ETH_Handler()
```

Comment:
You can register your own callback via NVIC setvector , we do not need to do weakly linking in for irq handler or?

Reply:
I assume `ETH_IRQHandler` is the one registered so an app can override this and do their own handling

Reply:
The requirement https://github.com/ARMmbed/mbed-os/issues/13554 is without any modification of the existing code tree , the user application needs to have its own handling, if **we use the NVIC_SetVector for the user application, again we need to modify the driver code**. Hence This proposal has given, in the file connectivity/drivers/emac/TARGET_STM/stm32xx_eth_irq_callback.cpp, we will have a #ifndef USE_USER_DEFINED_HAL_ETH_IRQ_CALLBACK flag which makes the default STM_HAL_ETH_Handler if the application doesn't have its callback.